surfactant/relationships/dotnet_relationship.py

```diff
@@ -65,6 +66,62 @@ def establish_relationships(
                                 dnProbingPaths = []
                             dnProbingPaths.append(pathlib.PureWindowsPath(path).as_posix())
 
+    # https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/loading-unmanaged
+    # 1. Check the active AssemblyLoadContext cache
+    # 2. Calling the import resolver set by the setDllImportResolver function
+    #    - a. Example using SetDllImportResolver: https://learn.microsoft.com/en-us/dotnet/standard/native-interop/native-library-loading
+    #    - b. Checks PInvoke's or Assembly's DefaultDllImportSearchPathsAttribute, then the assembly's directory, then LoadLibraryEx with LOAD_WITH_ALTERED_SEARCH_PATH flag
+    #       - This attribute has no effect on non-Windows platforms / Mono runtime
+    #       - https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.defaultdllimportsearchpathsattribute?view=net-7.0
+    #       - i. This has a "Paths" property which is a bitwise combination of paths specified in ii:
+    #       - ii. https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.dllimportsearchpath?view=net-7.0
+    # 3. The active AssemblyLoadContext calls its LoadUnmanagedDll function (Default behavior is the same as AssemblyRef probing?)
+    #    - a. Can be overridden, but the default implementation returns IntPtr.Zero, which tells the runtime to load with its default policy.
+    #    - b. https://learn.microsoft.com/en-us/dotnet/api/system.runtime.loader.assemblyloadcontext.loadunmanageddll?view=net-7.0
+    # 4. Run default unmanaged library probing logic by parsing *.deps.json probing properties
+    #    - a. If the json file isn't present, assume the calling assembly's directory contains the library
+    #    - b. https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/default-probing#unmanaged-native-library-probing
+    if "dotnetImplMap" in metadata:
+        for asmRef in metadata["dotnetImplMap"]:
+            if "Name" not in asmRef:
+                continue
+            refName = asmRef["Name"]
+
+            # Check absolute path against entries in software
+            if is_absolute_path(refName):
+                ref_abspath = pathlib.PureWindowsPath(refName)
+                for e in sbom.software:
+                    if e.installPath is None:
+                        continue
+                    if isinstance(e.installPath, Iterable):
+                        for ifile in e.installPath:
+                            if ref_abspath == pathlib.PureWindowsPath(ifile):
+                                relationships.extend(Relationship(dependent_uuid, e.uuid, "Uses"))
+                relationships.extend(get_windows_pe_dependencies(sbom, software, [refName]))
```

Comment:
I don't think the line calling `get_windows_pe_dependencies` should be present - haven't confirmed, but in that function it is appending refName (which in this case is an absolute path such as C:/Somefolder/Dep.dll) with the path software is located in, so it will just be searching for garbage paths like C:/PrgFiles/C:/Somefolder/Dep.dll that are actually invalid path names on Windows (: is not allowed in folder names) so the files it is looking for should never exist.